ui/App.jsx

```diff
@@ -102,9 +110,6 @@ const App = () => {
               path="/taskcluster-auth"
               render={(props) => <TaskclusterCallback {...props} />}
             />
-            <Route exact path="/">
-              <Redirect to="/jobs" />
-            </Route>
```

Comment:
For some reason, after I removed the extra `BrowserRouter` in index.jsx - which you're not supposed to have when using ConnectedRouter since it causes navigation issues - this redirect was overriding the `updateOldUrls` so I removed it and added a condition to handle the default path.

 That's why the backwards-compatibility was not working Monday. Murphy's law that a seemingly innocuous change that I didn't test for caused another bug. Anyhow, I've added a few tests.